pkg/kubelet/container/runtime.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/volume"
+)
+
+// Runtime interface defines the interfaces that should be implemented
+// by a container runtime.
+type Runtime interface {
+	// Version returns a map of version information of the container runtime.
+	Version() (map[string]string, error)
+	// ListPods returns a list of Pod. The boolean parameter specifies whether
+	// the runtime returns all pods including those already exited and dead pods(used
+	// for garbage collection).
+	ListPods(all bool) ([]*api.Pod, error)
+	// RunPod starts all the containers of a pod within a namespace.
+	RunPod(*api.Pod, map[string]volume.Interface) error
+	// KillPod kills all the containers of a pod.
+	KillPod(*api.Pod) error
+	// RunContainerInPod starts a container within the same namespace of a pod.
+	RunContainerInPod(api.Container, *api.Pod, map[string]volume.Interface) error
+	// KillContainerInPod kills a container in the pod.
+	KillContainerInPod(api.Container, *api.Pod) error
```

Comment:
My idea is `GetPodStatus()` is automatically done during `ListPods()`. So we don't need call it multiple times. 

But I think we still need a GetPodStatus in kubelet, because it needs to read the podstatus cache and needs to set the readiness.
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1982
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubelet/kubelet.go#L2018